Core/HttpHandlerProxy.cs

```diff
@@ -0,0 +1,151 @@
+
+        /// <summary>
+        /// Gets or sets proxy information for the request.
+        /// </summary>
+        public IWebProxy Proxy { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that indicates whether to send an authenticate header with the request.
+        /// </summary>
+        /// <returns>true to send a WWW-authenticate HTTP header with requests after authentication has taken place; otherwise, false. The default is false.</returns>
+        public bool PreAuthenticate { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="T:System.Boolean"/> value that controls whether default credentials are sent with requests.
+        /// </summary>
+        /// <returns>true if the default credentials are used; otherwise false. The default value is false.</returns>
+        public bool UseDefaultCredentials { get; set; }
+
+        public HttpHandlerProxy(IHttpMessageHandlerFactory httpMessageHandlerFactory, string httpClientName)
+        {
+            this.httpClientName = httpClientName;
+            this.httpMessageHandler = httpMessageHandlerFactory?.CreateHandler(httpClientName);
+            this.httpClientHandler = null == httpMessageHandlerFactory ? HttpClientExtensions.CreateHttpClientHandler(httpClientName) : null;
+        }
+
+        private void Authorize(HttpRequestMessage request)
+        {
+            switch (Credentials)
+            {
+                case NetworkCredential networkCredential:
+                    {
+                        string authString = !string.IsNullOrEmpty(networkCredential.Domain)
+                            ? networkCredential.Domain + "\\" + networkCredential.UserName + ":" + networkCredential.Password
+                            : networkCredential.UserName + ":" + networkCredential.Password;
+
+                        string base64AuthString = Convert.ToBase64String(Encoding.UTF8.GetBytes(authString));
+
+                        request.Headers.Authorization = new AuthenticationHeaderValue("Basic", base64AuthString);
+                    }
+                    break;
+
+                case null:
+                    break;
+
+                default:
+                    throw new AuthenticationException($"Credentials of type '{Credentials.GetType().Name}' is not supported");
+            }
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var cookieHeader = CookieContainer?.GetCookieHeader(request.RequestUri);
+            if (!string.IsNullOrEmpty(cookieHeader))
+            {
+                request.Headers.Add(HeaderNames.Cookie, cookieHeader);
+            }
+
+            Authorize(request);
+
+            HttpResponseMessage response = null;
+            if (null != InnerHandler)
+            {
+                response = await base.SendAsync(request, cancellationToken);

```

Comment:
If you initialized InnerHandler, and set Credentials for it, do you need to keep calling Authorize() on it? It may break logic if you both authorize with Basic in Authorize(), and with NTLM/Digest by handler.Credentials internally.

I'd move this to line 111 where you probe if shared handler can work with EWS using BASIC.

Finally, we don't know how many of EWS endpoints will accept basic. I'd log it somewhere where we can query AppInsights later and know

Reply:
1. Authorize() and negotiate, agree that it is not good to set Authenticate header one negotiate took place, but I have checked this, it works well, any way will remove Authorize in after negotiate.
2. I think none, during testing it was always negotiate, and basic is absolutely insecure. May be it is worth just remove logic with probe for basic auth.

Reply:
1. Even if works now may not work later - I'd call Authorize() only for probe call..
2. Or indeed if BASIC probing has no value - always use private handler whever Credentials is set
3. In your code, is there scenario where same EWS URL + Creds are used in parallel (impersonated access to exchange and scan of multiple mailboxes in tenant?) - if so there COULD be benefit of caching DelegatingHandler per (EWS URL + Credentials object) - but this may be hard to do right...